framework/src/timesteppers/CompositionDT.C

```diff
@@ -0,0 +1,178 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CompositionDT.h"
+#include "MooseApp.h"
+#include "Transient.h"
+#include "TimeSequenceStepperBase.h"
+
+registerMooseObject("MooseApp", CompositionDT);
+
+InputParameters
+CompositionDT::validParams()
+{
+  InputParameters params = TimeStepper::validParams();
+
+  params.addRequiredParam<std::string>(
+      "base_timestepper", "Provide a time stepper type to produce a base time step size");
+
+  params.addParam<std::vector<std::string>>(
+      "input_timesteppers",
+      "The input TimeSteppers to compose time step size.  This can either be N timesteppers or 1 "
+      "timestepper.");
+
+  params.addParam<MooseEnum>(
+      "composition_type",
+      getCompositionTypes(),
+      "Provide a compose method to operate on input TimeSteppers. The composed time step size "
+      "provide a max/min value for the current time step size. Avaliable methods includes max "
+      "step, min step");
+
+  params.addParam<std::string>("times_to_hit_timestepper",
+                               "Provide user specified time to hit with a time sequence stepper");
+
+  params.addParam<Real>("dt", "Initial value of dt");
+
+  params.addParamNamesToGroup("input_timesteppers composition_type",
+                              "Time Steppers for composition");
+
+  params.addParamNamesToGroup("times_to_hit_timestepper", "Time Sequence Stepper");
+
+  params.addClassDescription("Compose multiple TimeSteppers together to generate time step size.");
+
+  return params;
+}
+
+CompositionDT::CompositionDT(const InputParameters & parameters)
+  : TimeStepper(parameters),
+    _has_initial_dt(isParamValid("dt")),
+    _initial_dt(_has_initial_dt ? getParam<Real>("dt") : 0.),
+    _base_timestepper(getParam<std::string>("base_timestepper"))
+{
+  if (isParamValid("input_timesteppers"))
+    _inputs = parameters.get<std::vector<std::string>>("input_timesteppers");
+
+  if (isParamValid("times_to_hit_timestepper"))
+    _hit_timestepper_name = parameters.get<std::string>("times_to_hit_timestepper");
+}
+
+Real
+CompositionDT::computeInitialDT()
+{
+  if (_has_initial_dt)
+    return _initial_dt;
+  else
+    return computeDT();
+}
+
+Real
+CompositionDT::computeDT()
+{
+  std::map<const std::string, Real> dts;
```

Comment:
why a map?

Reply:
I was think if we need to output the current TimeStepper name used for the current time step(maybe in debug mode). Since we have multiple TimeSteppers and it may be easier for the users to understand the time stepper behavior. Any thoughts?

Reply:
If you add this, sure. Without this, it could just be a vector. 

You ll need to pull the Problem verbose parameter, which already gives timestepper information